### ISSUE TITLE
Using the string literal "this" will not lead to a tslint error when accessing a dictionary

### ISSUE THREAD

**Issue by mtennoe** [palantir/tslint#809]
A minimal repro is:

``` javascript
const dictionary: {[key: string]: string} = null;
dictionary["this"] = "";
dictionary["notThis"] = "";
```

The first line of accessing the dictionary, using "this", will not throw a tslint error, while the second one will throw the expected "object access via string literals is disallowed". The usage of "this" should also throw this error.

**Comment by ajafff** [comment]
This is fixed by #2495
